simulation_ws/src/osr_urdf_with_controllers_and_launch/rover_gazebo/launch/rover.launch

```diff
@@ -30,12 +30,10 @@
 
       <!-- convert joint states to TF transforms for rviz, etc -->
       <remap from="joint_states" to="rover_ns/joint_states" />
-      <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
-        respawn="true" output="screen">
-      </node>
+      
+      <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true" output="screen" />
```

Comment:
Might be better to let the node die instead of respawning which tends to hide the issue.